tempodb/backend/azure/azure.go

```diff
@@ -292,76 +294,74 @@ func (rw *readerWriter) append(ctx context.Context, src []byte, name string) err
 		return blockIDBinaryToBase64(binaryBlockID)
 	}
 
-	l, err := appendBlobURL.GetBlockList(ctx, blob.BlockListAll, blob.LeaseAccessConditions{})
+	l, err := appendBlobClient.GetBlockList(ctx, blockblob.BlockListTypeAll, &blockblob.GetBlockListOptions{})
 	if err != nil {
 		return err
 	}
 
 	// generate the next block id
 	id := blockIDIntToBase64(len(l.CommittedBlocks) + 1)
 
-	_, err = appendBlobURL.StageBlock(ctx, id, bytes.NewReader(src), blob.LeaseAccessConditions{}, nil, blob.ClientProvidedKeyOptions{})
+	_, err = appendBlobClient.StageBlock(ctx, id, streaming.NopCloser(bytes.NewReader(src)), &blockblob.StageBlockOptions{})
 	if err != nil {
 		return err
 	}
 
 	base64BlockIDs := make([]string, len(l.CommittedBlocks)+1)
 	for i := 0; i < len(l.CommittedBlocks); i++ {
-		base64BlockIDs[i] = l.CommittedBlocks[i].Name
+		base64BlockIDs[i] = *l.CommittedBlocks[i].Name
 	}
 
 	base64BlockIDs[len(l.CommittedBlocks)] = id
 
 	// After all the blocks are uploaded, atomically commit them to the blob.
-	_, err = appendBlobURL.CommitBlockList(ctx, base64BlockIDs, blob.BlobHTTPHeaders{}, blob.Metadata{}, blob.BlobAccessConditions{}, blob.DefaultAccessTier, blob.BlobTagsMap{}, blob.ClientProvidedKeyOptions{}, blob.ImmutabilityPolicyOptions{})
+	_, err = appendBlobClient.CommitBlockList(ctx, base64BlockIDs, &blockblob.CommitBlockListOptions{})
 	if err != nil {
 		return err
 	}
 	return nil
 }
 
 func (rw *readerWriter) writer(ctx context.Context, src io.Reader, name string) error {
-	blobURL := rw.containerURL.NewBlockBlobURL(name)
+	blobClient := rw.containerClient.NewBlockBlobClient(name)
 
-	if _, err := blob.UploadStreamToBlockBlob(ctx, src, blobURL,
-		blob.UploadStreamToBlockBlobOptions{
-			BufferSize: rw.cfg.BufferSize,
-			MaxBuffers: rw.cfg.MaxBuffers,
-		},
-	); err != nil {
+	if _, err := blobClient.UploadStream(ctx, src, &azblob.UploadStreamOptions{
+		BlockSize:   int64(rw.cfg.BufferSize),
+		Concurrency: rw.cfg.MaxBuffers,
+	}); err != nil {
 		return errors.Wrapf(err, "cannot upload blob, name: %s", name)
 	}
 	return nil
 }
 
 func (rw *readerWriter) readRange(ctx context.Context, name string, offset int64, destBuffer []byte) error {
-	blobURL := rw.hedgedContainerURL.NewBlockBlobURL(name)
+	blobClient := rw.hedgedContainerClient.NewBlockBlobClient(name)
 
-	var props *blob.BlobGetPropertiesResponse
-	props, err := blobURL.GetProperties(ctx, blob.BlobAccessConditions{}, blob.ClientProvidedKeyOptions{})
+	props, err := blobClient.GetProperties(ctx, &blob.GetPropertiesOptions{})
 	if err != nil {
 		return err
 	}
 
 	length := int64(len(destBuffer))
 	var size int64
 
-	if length > 0 && length <= props.ContentLength()-offset {
+	if length > 0 && length <= *props.ContentLength-offset {
```

Comment:
similar pointer concerns here

Reply:
Action taken: we now return an error if `props.ContentLength == nil`.